ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStoreV2.java

```diff
@@ -235,17 +237,67 @@ public void updateTtl(List<MessageInfo> messageInfos) {
     // TODO
   }
 
+  /**
+   * Finds keys absent in the Store from given list of keys
+   * @param storeKeys Input list of keys to check for existence
+   * @return Set of keys absent in the Store
+   */
   @Override
   public Set<StoreKey> findMissingKeys(List<StoreKey> storeKeys) {
-    // TODO
-    return new HashSet<>();
+    Set<StoreKey> missingKeys = new HashSet<>();
+    Timer.Context timer = azureMetrics.missingKeysQueryTime.time();
+    try {
+      for (StoreKey storeKey : storeKeys) {
+        try {
+          if (!blobContainerClient.getBlobClient(storeKey.getID()).exists()) {
```

Comment:
Is there any batched API to check the existence for multiple blobs in Azure?

Reply:
Good question ! I looked for this first and there is no such API
https://stackoverflow.com/questions/54879762/how-to-check-if-multiple-files-exists-in-azure-container